conf/modules.config

```diff
@@ -0,0 +1,770 @@
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            enabled: params.save_intermediates
+        ]
+    }
+
+    withName: DCC_MATE2_2ND_PASS {
+        ext.when = { !meta.single_end && params.module.split(',').contains('circrna_discovery') && params.tool.split(',').contains('dcc') }
+        ext.prefix = { "${meta.id}_mate2" }
+        ext.args = [ "",
+            "--chimOutType Junctions WithinBAM",
+            "--outSAMunmapped Within",
+            "--outFilterType BySJout",
+            "--outReadsUnmapped None",
+            "--readFilesCommand zcat",
+            "--sjdbFileChrStartEnd dataset.SJ.out.tab",
+            params.alignSJDBoverhangMin ? "--alignSJDBoverhangMin ${params.alignSJDBoverhangMin}" : "--alignSJDBoverhangMin 10",
+            params.chimJunctionOverhangMin ? "--chimJunctionOverhangMin ${params.chimJunctionOverhangMin}" : "--chimJunctionOverhangMin 10",
+            params.chimSegmentMin ? "--chimSegmentMin ${params.chimSegmentMin}" :  "--chimSegmentMin 10"
+        ].join(' ').trim()
+        publishDir = [
+            path: { "${params.outdir}/circrna_discovery/dcc/intermediates/mate2/2nd_pass" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            enabled: params.save_intermediates
+        ]
+    }
+
+    withName: DCC {
+        ext.when = { params.tool.split(',').contains('dcc') && params.module.split(',').contains('circrna_discovery') }
+        publishDir = [
+            path: { "${params.outdir}/circrna_discovery/dcc/intermediates/" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            enabled: params.save_intermediates
+        ]
+    }
+
+    withName: MAPSPLICE_REFERENCE {
+        ext.when = { params.module.split(',').contains('circrna_discovery') && params.tool.split(',').contains('mapsplice') }
+        ext.args = [ "",
+            "-genePredExt",
+            "-geneNameAsName2"
+        ].join(' ').trim()
+        publishDir = [
+            path: { "${params.outdir}/references/mapsplice" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            enabled: params.save_reference
+        ]
+    }
+
+    withName: MAPSPLICE_ALIGN {
+        ext.when = { params.module.split(',').contains('circrna_discovery') && params.tool.split(',').contains('mapsplice') }
+        ext.args = [ "",
+            params.seglen ? "--seglen ${params.seglen}" : "--seglen 25",
+            params.min_intron ? "--min-intron ${params.min_intron}" : "--min-intron 20",
+            params.max_intron ? "--max-intron ${params.max_intron}" : "--max-intron 1000000",
+            params.min_map_len ? "--min-map-len ${params.min_map_len}" : "--min-map-len 40",
```

Comment:
Could these not just be defined by the `defaults` in nextflow.config, rather than having the if statement?

Reply:
Same applies elsehwere